src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3058,16 +3028,6 @@ bool ShouldWarnForReflectionAccessToCompilerGeneratedCode (MethodDefinition meth
 
 		void ProcessAnalysisAnnotationsForMethod (MethodDefinition method, DependencyKind dependencyKind, in MessageOrigin origin)
 		{
-			// There are only two ways to get there such that the origin isn't the same as the top of the scopestack.
-			// - For DAM on type, the current scope is the caller of GetType, while the origin is the type itself.
-			// - For warnings produced inside compiler-generated code, the current scope is the user code that
-			//   owns the compiler-generated code, while the origin is the compiler-generated code.
-			// In either case any warnings produced here should use the origin instead of the scopestack.
-			if (origin.Provider != ScopeStack.CurrentScope.Origin.Provider) {
-				Debug.Assert (dependencyKind == DependencyKind.DynamicallyAccessedMemberOnType ||
-					(origin.Provider is MethodDefinition originMethod && CompilerGeneratedState.IsNestedFunctionOrStateMachineMember (originMethod)));
-			}
-
```

Comment:
This is similar to above with ProcessAnalysisAnnotationsForField. Since we no longer use ScopeStack, this can be removed.